### PR TITLE
docs: Adding auto generation from source

### PR DESCRIPTION
**Pull request by hammenm**
# Pull Request Template

## PR Checklist

- [x] I have run `npm test` locally and all tests are passing.
- [x] I have added/updated tests for any new behavior. - *No test has been added*
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: This issue has been refered in issue #64 

## PR Description

A package has been added to auto generate docs from source [jsdoc](https://github.com/jsdoc3/jsdoc). To be noted:
- The destination folder of the documentation is in `docs/generated`.
- Some warning appear during the generation. It should be fixed directly in the code.
- Ignore files has been updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "lint": "eslint .",
     "lintfix": "eslint --fix lib/**",
     "test": "eslint lib/** && jest --coverage --maxWorkers=4",
-    "coveralls": "jest --coverage && cat ./tests/coverage/lcov.info | coveralls"
+    "coveralls": "jest --coverage && cat ./tests/coverage/lcov.info | coveralls",
+    "doc": "jsdoc -r lib -d docs/generated -R README.md"
```
Comment (vvscode) [inline package.json]: it worth add `jsdoc` to dependencies list. Currently it would work only with global jsdoc

Comment (jesus-seijas-sp) [reply package.json]: I agree with @vvscode, jasdoc should be a dev dependency.

**Comment by vvscode** [comment]
@hammenm  fix conflicts, please

**Comment by jesus-seijas-sp** [comment]
This PR is no longer needed as another PR already covered the need, with jsdoc in dependencies and uploading the docs generated.

**Pull request closed without merging**